src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -200,40 +196,69 @@ public static NodeSettings Default(Network network = null, ProtocolVersion proto
-        /// <returns>Initialized node configuration.</returns>
-        /// <exception cref="ConfigurationException">Thrown in case of any problems with the configuration file or command line arguments.</exception>
-        public NodeSettings LoadConfiguration(List<IFeatureRegistration> features = null)
+        /// <returns>Returns <c>true</c> if the file was created and <c>false</c> otherwise.</returns>
+        public bool CreateDefaultConfigurationFile(List<IFeatureRegistration> features)
         {
-            // Configuration already loaded?
-            if (this.ConfigReader != null)
-                return this;
+            // If the config file does not exist yet then create it now.
+            if (this.ConfigurationFile != null && !File.Exists(this.ConfigurationFile))
+            {
+                this.Logger.LogDebug("Creating configuration file '{0}'.", this.ConfigurationFile);
 
-            // Get the arguments set previously
-            var args = this.LoadArgs;
+                File.WriteAllText(this.ConfigurationFile, this.GetDefaultConfiguration(features));
+                this.SetCombinedConfiguration();
+                this.LoadConfiguration();
 
-            // If no configuration file path is passed in the args, load the default file.
-            if (this.ConfigurationFile == null)
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Sets the ConfigReader based on the arguments and the configuration file if it exists.
+        /// Also sets the logging settings for the configuration.
+        /// </summary>
+        private void SetCombinedConfiguration()
+        {
+            // Read the configuration file if it exists.
+            string configText = "";
+            if (this.ConfigurationFile != null && File.Exists(this.ConfigurationFile))
             {
-                this.ConfigurationFile = this.CreateDefaultConfigurationFile(features);
+                this.Logger.LogDebug("Reading configuration file '{0}'.", this.ConfigurationFile);
+                configText = File.ReadAllText(this.ConfigurationFile);
             }
+            else
+                this.Logger.LogDebug("No configuration file. Proceeding with defaults.");
+
+            // Get the arguments set previously.
+            var args = this.LoadArgs;
 
             // Add the file configuration to the command-line configuration.
-            var fileConfig = new TextFileConfiguration(File.ReadAllText(this.ConfigurationFile));
+            var fileConfig = new TextFileConfiguration(configText);
             var config = new TextFileConfiguration(args);
-            this.ConfigReader = config;
             fileConfig.MergeInto(config);
 
-            // Set the configuration filter and file path.
+            // Set logging settings for this configuration.
```

Comment:
Is this the right place?
This method does a bit more than it says on the tin.

Reply:
The method description is as follows:
```
        /// <summary>
        /// Sets the ConfigReader based on the arguments and the configuration file if it exists.
        /// Also sets the logging settings for the configuration.
        /// </summary>
        private void SetCombinedConfiguration()
        {
```
It does say: `Also sets the logging settings for the configuration.`

Reply:
If we remove this code from `SetCombinedConfiguration` then we would need to call it everywhere we call this method. An alternative name for this method could be `OnConfigurationChanged` because it does perform any action needed after the configuration file may have changed. `LoadConfiguration` is taking on a role more focused on actually parsing/reading the node settings from the `ConfigReader`.

`this.Log.Load(config)` could move to `LoadConfiguration` but it is probably beneficial to have it being called earlier in the constructor by virtue of the call to the first `SetCombinedConfiguration`.

Reply:
Refactored the logging related code.